### PR TITLE
Use new /sign/file autograph endpoint

### LINKED ISSUE #9308: sign addons with autograph /sign/file
Per https://github.com/mozilla/addons-server/pull/9306#issuecomment-417300984, we want to switch addons-server to the autograph [`/sign/file` endpoint](https://github.com/mozilla-services/autograph/tree/6bc03e3ce0ce8aae6e31b8240159ec2c7bceef86/signer/xpi#signature-request).

This will let us eventually migrate to SHA256 digests for the PKCS7 signature and add COSE Signatures for https://bugzilla.mozilla.org/show_bug.cgi?id=1471186

We'll probably want to:

- [ ] QA the autograph-signed addons (this probably entails fixing ~~https://github.com/mozilla-services/autograph/issues/130~~ (edit: not necessary; closed) and https://github.com/mozilla-services/autograph/issues/122 and making sure autograph XPIs pass other signing-clients and addons-server unit tests)  
- [ ] determine peak load and network capacity and load test autograph

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #9308

Various cleanups along the way, this PR is porting some code over from `signing-clients` so that we'll be able to remove that dependency once we don't have to do the actual signing on our end anymore.

I'm also renaming `crypto/packaged.py` into `crypto/signing.py` since it feels more fitting.

### CODE REVIEW

File: src/olympia/lib/crypto/signing.py [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+                'pkcs7_digest': 'SHA1',
+                'cose_algorithms': ['ES256']
+            },
+        }]
+
+        with statsd.timer('services.sign.addon.autograph'):
+            response = requests.post(
+                '{server}/sign/file'.format(server=conf['server_url']),
+                json=signing_request,
+                auth=HawkAuth(id=conf['user_id'], key=conf['key']))
```
Comment (EnTeQuAk) [inline src/olympia/lib/crypto/signing.py]: cc @g-k to just double-check the code above and that the docs are right.

Comment (g-k) [reply src/olympia/lib/crypto/signing.py]: :+1: lgtm

Also, not sure how much detail you want in include about signature validation in the docs, but:

* the relevant pref is "security.signed_app_signatures.policy"
* in Fx nightly (and stable I believe) it's currently set to 2 for COSEAndPKCS7WithSHA1OrSHA256 in https://searchfox.org/mozilla-central/rev/dac799c9f4e9f5f05c1071cba94f2522aa31f7eb/security/manager/ssl/tests/unit/test_signed_apps.js#190-198

**Comment by EnTeQuAk** [comment]
This PR is ready for a first look but realizing that CircleCI tests are failing I think I'd like to fix the CircleCI tests via https://github.com/mozilla/addons-server/issues/10438 - or a separate PR if I realize that this is turning into too much work.

**Comment by EnTeQuAk** [comment]
https://github.com/mozilla/addons-server/issues/10438 will be fixed via #10467 so this isn't part of this PR

File: src/olympia/lib/crypto/signing.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+
+def call_signing(file_obj):
+    """Get the jar signature and send it to the signing server to be signed."""
+    # Extract jar signature.
+    conf = settings.AUTOGRAPH_CONFIG
+    log.debug('Calling autograph service: {0}'.format(
+        conf['server_url']))
+
+    if waffle.sample_is_active('activate-autograph-file-signing'):
```
Comment (diox) [inline src/olympia/lib/crypto/signing.py]: Because of the waffle this `call_signing()` function is now super long and hard to follow. Can you break it down?

Comment (diox) [reply src/olympia/lib/crypto/signing.py]: Also, add a migration to add the default values for the waffle in the database ?

Comment (EnTeQuAk) [reply src/olympia/lib/crypto/signing.py]: Fixed in 4590507769, I hope that's a bit easier to read?

**Comment by EnTeQuAk** [comment]
Actually, some documentation about this: https://github.com/mozilla-services/autograph/blob/master/signer/xpi/README.rst#signature-request

File: src/olympia/lib/crypto/signing.py [hunk 3]
```diff
@@ -53,19 +55,79 @@ def get_id(addon):
...
-    """Get the jar signature and send it to the signing server to be signed."""
-    # Extract jar signature.
+def autograph_sign_file(file_obj):
+    """Sign `file_obj` via autographs /sign/file endpoint.
+
+    :returns: The certificates serial number.
+    """
+    conf = settings.AUTOGRAPH_CONFIG
+
+    with storage.open(file_obj.current_file_path) as fobj:
```
Comment (diox) [inline src/olympia/lib/crypto/signing.py]: You should open in `rb` mode I think since you want to encode in base64, which takes bytes.

Comment (diox) [reply src/olympia/lib/crypto/signing.py]: Then again this probably doesn't matter in Py2 and you've got more potential for breakage through other calls below that deal with bytes in various ways... So maybe don't bother for now, it will have to be fixed in https://github.com/mozilla/addons-server/issues/10423

Comment (EnTeQuAk) [reply src/olympia/lib/crypto/signing.py]: Perfect, yeah, happy to tackle that as part of #10423 then
